Daniel_Heffner/angular/product_factory/scriptular.js

```diff
@@ -0,0 +1,76 @@
+var appModule = angular.module('product_app', []);
+
+appModule.factory('product_factory', function(){
+	var products = [
+		{name: 'Sample Product', price: 39.95},
+	];
+	var factory = {}
+	factory.getProducts = function(callback){
+		callback(products);
+		return false;
+	}
+	factory.addProduct = function(name, price){
+		if (name != '' && (price != '' && price > 0)){
+			products.push({name: name, price: price});
+			var errors = {success: true};
+			return errors;
+		} else {
+			var errors = {success: false};
+			if (name == '') errors.name = 'Products must have names!';
+			if (price == '' || !(price > 0)) errors.price = 'Products must cost something!';
+			return errors;
+		}
+	}
+	factory.destroyProduct = function(idx){
+		console.log('sanity check');
+		products.splice(idx, 1);
+	}
+	return factory;
+})
+
+appModule.controller('product_controller', ['$scope', 'product_factory', function($scope, product_factory){
+	$scope.products = [];
+	product_factory.getProducts(function(data){
+		$scope.products = data;
+	});
+	$scope.newName = '';
```

Comment:
I would put all of the `$scope.newX` variables into `$scope.new.X` -- so that clearing them all is as easy as `$scope.new = null;`